src/data/ability.ts

```diff
@@ -2106,6 +2107,22 @@ export class PostTurnFormChangeAbAttr extends PostTurnAbAttr {
   }
 }
 
+export class FetchBallAbAttr extends PostTurnAbAttr {
+  constructor() {
+    super();
+  }
+  applyPostTurn(pokemon: Pokemon, passive: boolean, args: any[]): boolean {
+    let lastUsed = pokemon.scene.currentBattle.lastUsedPokeball;
+    if(lastUsed != null && pokemon.isPlayer) {
```

Comment:
I believe you can check if lastUsed is null with just
```suggestion
    if(!lastUsed && pokemon.isPlayer) {
```

Reply:
Nah, I think that's backwards. You would remove the exclamation mark.

Reply:
I think you're right. Can't believe I made such a typo 🤦